tests/modes.spec.ts

```diff
@@ -54,6 +54,31 @@ test("creating a new freehand polygon opens a form", async () => {
   await page.getByLabel("Description:").click();
 });
 
+test("creating a new freehand polygon and switching modes doesn't lose changes", async () => {
+  await page.getByRole("button", { name: "New polygon (freehand)" }).click();
+  await clickMap(page, 500, 500);
+  await clickMap(page, 400, 500);
+  await clickMap(page, 400, 600);
+
+  // Switch modes without finishing
+  await page.getByRole("button", { name: "New point" }).click();
+
+  await page.getByRole("button", { name: "1) Untitled area" }).click();
+  await page.getByLabel("Description:").click();
+});
+
+test("creating a new freehand polygon and canceling doesn't save anything", async () => {
+  await page.getByRole("button", { name: "New polygon (freehand)" }).click();
+  await clickMap(page, 500, 500);
+  await clickMap(page, 400, 500);
+  await clickMap(page, 400, 600);
+
+  await page.getByRole("button", { name: "Cancel" }).click();
+  await expect(page.getByText("0 objects")).toBeVisible();
+});
+
+// TODO Repeat switching modes and canceling for other draw tools
```

Comment:
We have really repetitive tests... it's tedious to enumerate all of them manually. Will think about future refactors to make this less painful.